pytorch_to_returnn/naming/naming.py

```diff
@@ -21,6 +21,7 @@ class Naming:
   modules: OrderedDict[_types.Module, _module.ModuleEntry]
   inputs: List[_tensor.TensorEntry]
   outputs: List[_tensor.TensorEntry]
+  non_deterministic_layer_outputs: Dict[str, numpy.ndarray] = {}
```

Comment:
You should **not** (never) make a mutable object instance (e.g. a dict) as a class attribute. Otherwise multiple `Naming` instances might share the same `non_deterministic_layer_outputs` which is not what you want. This should be assigned in the `__init__`. Only the type annotation can stay here.